bin/python_scripts/find_invalid_tags.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Set up environment from ckan.ini
+# export POSTGRES_URL=<sqlalchemy.url from ckan.ini>
+#
+# Execute script like this -
+# python find_invalid_tags.py
+#
+
+import io
+import os
+import sys
+import psycopg2
+import subprocess
+
+import logging
+
+from ckan.logic.schema import default_tags_schema
+from ckan.lib.navl.dictization_functions import validate, Invalid
+
+POSTGRES_URL = os.environ.get('POSTGRES_URL')
+
+logger = logging.getLogger(__name__)
+connection = psycopg2.connect(POSTGRES_URL)
+
+
+def setup_logging():
+    _format = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    logger.setLevel(logging.INFO)
+    c_handler = logging.StreamHandler()
+    c_handler.setFormatter(_format)
+    logger.addHandler(c_handler)
+
+    f_handler = logging.FileHandler('find_invalid_tags.log')
+    f_handler.setFormatter(_format)
+    logger.addHandler(f_handler)
+
+    logger.info('====================================================================')
+
+
+# only target packages with tags that are active
+def get_all_tags():
+    cursor = connection.cursor()
+    sql = """
+    SELECT tag.name, count(tag.name) FROM tag, package, package_tag
+    WHERE tag.id = package_tag.tag_id AND package.id = package_tag.package_id AND package.state = 'active'
+    GROUP BY tag.name ORDER BY count(tag.name) DESC;
+    """
+
+    cursor.execute(sql)
+    return cursor
+
+
+def create_tag_list(rows):
+    with io.open("invalid_tags.txt", "w+", encoding="utf-8") as f:
+        for row in rows:
+            f.write(row)
```

Comment:
I think `rows` is coming in as a string to this function, so it's writing each character one by one. `f.write(rows)` on its own should be the same, although I think it's probably better if we're passing a `list` into this function.

Reply:
Thats true